src/System.Data.Common/tests/System/Data/Common/DbCommandTest.cs

```diff
@@ -0,0 +1,105 @@
+                new FinalizingCommand();
+            }
+
+            protected override void Dispose(bool disposing)
+            {
+                if (!disposing)
+                    _wasFinalized = true;
+                base.Dispose(disposing);
+            }
+
+            public override string CommandText
+            {
+                get => throw new NotImplementedException();
+                set => throw new NotImplementedException();
+            }
+
+            public override int CommandTimeout
+            {
+                get => throw new NotImplementedException();
+                set => throw new NotImplementedException();
+            }
+
+            public override CommandType CommandType
+            {
+                get => throw new NotImplementedException();
+                set => throw new NotImplementedException();
+            }
+
+            public override bool DesignTimeVisible
+            {
+                get => throw new NotImplementedException();
+                set => throw new NotImplementedException();
+            }
+
+            public override UpdateRowSource UpdatedRowSource
+            {
+                get => throw new NotImplementedException();
+                set => throw new NotImplementedException();
+            }
+
+            protected override DbConnection DbConnection
+            {
+                get => throw new NotImplementedException();
+                set => throw new NotImplementedException();
+            }
+
+            protected override DbParameterCollection DbParameterCollection => throw new NotImplementedException();
+            protected override DbTransaction DbTransaction
+            {
+                get => throw new NotImplementedException();
+                set => throw new NotImplementedException();
+            }
+
+            public override void Cancel() => throw new NotImplementedException();
+            public override int ExecuteNonQuery() => throw new NotImplementedException();
+            public override object ExecuteScalar() => throw new NotImplementedException();
+            public override void Prepare() => throw new NotImplementedException();
+            protected override DbParameter CreateDbParameter() => throw new NotImplementedException();
+            protected override DbDataReader ExecuteDbDataReader(CommandBehavior behavior) => throw new NotImplementedException();
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Mono, "GC has different behavior on Mono")]
+        public void CanBeFinalized()
+        {
+            FinalizingCommand.CreateAndRelease();
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+            Assert.True(_wasFinalized);
+        }
+
+        [Fact]
+        public Task ExecuteReaderAsyncCanceled1()
+            => Assert.ThrowsAsync<TaskCanceledException>(() => new FinalizingCommand().ExecuteReaderAsync(new CancellationToken(true)));
```

Comment:
With this set up, you don't know whether the exception is being thrown synchronously out of ExecuteReaderAsync or whether ExecuteReaderAsync is canceling the returned task. Presumably you want to verify that the latter is happening, in which case it would be better for this to be:
```C#
[Fact]
public async Task ExecuteReaderAsyncCanceled()
{
    Task t = new FinalizingCommand().ExecuteReaderAsync(new CancellationToken(true));
    await Assert.ThrowsAsync<TaskCanceledException>(() => t);
}
```
Note, too, that while in production code you might want to optimize to avoid the extra async method, in tests that's not necessary, and it can actually be better in tests to await, because then the call stack will reflect which test method it came from.

Reply:
I see... I was under the impression nunit checked precisely for that but that seems to be wrong.

Will make the changes right away.

Reply:
Done